src/adapter/src/catalog.rs

```diff
@@ -2797,8 +2797,12 @@ impl<S: Append> Catalog<S> {
                         Op::UpdateSystemConfiguration { name, value }
                     }))
                     .collect::<Vec<_>>();
-
-                self.transact(None, None, ops, |_| Ok(())).await.unwrap();
+                // It is safe to pass in the minimum timestamp as the oracle
```

Comment:
This call site is the scary one and using MIN here seems dangerous. If we do billing based on the timestamp, can this ever cause users to be billed for 50 years of the default cluster size?

Reply:
Well, _this_ callsite is just about adding some initial system catalog parameters. So none of these will generate audit events.

That reminds me, though... we bootstrap those initial cluster events with the system clock's now: https://github.com/MaterializeInc/materialize/blob/7b6a392fa402a8a4f3269bbc0e44492f15a80a67/src/adapter/src/catalog/storage.rs#L184. Seems like we need to guarantee that we initialize the timestamp oracle to something no smaller than that value of now? In case of a very rare but possible backwards time jump between when we compute now in environmentd's main function and when we compute now when initializing the timestamp oracle?

If we plumb a "bootstrap now" parameter through all those places, then it probably makes sense to just plumb it in here, too!

Reply:
Yes! Seems like the initial migration should also populate the durable timestamps collection with that. cc @jkosh44

Reply:
Oh, neat, that makes sense.

Reply:
I think there is actually a small issue.

`bootstrap_args.now` is initialized here:
https://github.com/MaterializeInc/materialize/blob/797f2f833f40c66b517f52e9c8985a8158fa6d3f/src/environmentd/src/lib.rs#L347-L359
It's initialized as the current value of the system clock (`config.now` is a reference to the system clock).

The timestamp oracle's timestamp is initialized here: 
https://github.com/MaterializeInc/materialize/blob/797f2f833f40c66b517f52e9c8985a8158fa6d3f/src/adapter/src/coord.rs#L1185-L1216
The TLDR is that it's initialized to the max of the current value of the system clock and the durable timestamp from the previous Coord.

There is a small possibility that the system clock goes backwards after populating `bootstrap_args.now` but before populating the timestamp oracle's timestamp. In this scenario, the audit event timestamp can go backwards.



As a solution I think the timestamp oracle should set the initial timestamp to the max of the time used in `bootstrap_args.now` and the previous durable timestamp.

Reply:
Sorry I misunderstood the previous comments, I thought you were all saying that there is no problem. I see now that you were saying that there is a problem. In that case I am in agreement, there is a problem.

Reply:
Once https://github.com/MaterializeInc/materialize/pull/17005 lands, we'll be able to use the bootstrap time here, and this callsite will no longer be scary!